enhancements/api-review/add-new-CRD-ImageSourceDigestPolicy-and-ImageSourceTagPolicy-to-config.openshift.io.md

```diff
@@ -0,0 +1,400 @@
+and a list of mirrors require digest specification, or have different boolean value.
+- [openshift/api/config/v1](https://github.com/openshift/api/tree/master/config/v1): definition the schema and API of the ImageSourceDigestPolicy and ImageSourceTagPolicy.
+- [openshift/client-go](https://github.com/openshift/client-go), [openshift/cluster-config-operator](https://github.com/openshift/cluster-config-operator/pull/220): rebase the openshift/api version in these repositories to apply the new CRD
+to the cluster.
+- [openshift/runtime-utils/pkg/registries](https://github.com/openshift/runtime-utils/tree/master/pkg/registries): helper functions to edit registries.conf.
+- [openshift/machine-config-operator](https://github.com/openshift/machine-config-operator): MCO needs watch for the ImageSourceDigestPolicy and ImageSourceDigestPolicy. The machine-config-operator/pkg/controller/container-runtime-config controller needs to operate the ImageSourceDigestPolicy and ImageSourceDigestPolicy CRDs.
+Converts the existing ImageContenSourcePolicy objects to objects of new CRD.
+- This [document](https://docs.google.com/document/d/11FJPpIYAQLj5EcYiJtbi_bNkAcJa2hCLV63WvoDsrcQ/edit?usp=sharing) keeps a list of components that use operator.openshift.io/v1alpha1 ImageContentSourcePolicy. Need to change those repositories to upgrade to ImageSourceDigestPolicy.
+
+#### Notes
+
+1. During the upgrade path, MCO can watch for both old CR ImageContentSourcePolicy and new CRs and create objects.
+
+2. The merge order of mirrors for the same source is deterministic by topological sorting:<br/>
+Order is preserving the relative order of the mirrors using topological sorting. A graph is formed using each mirror as a
+vertex, adding an unidirectional edge between adjacent mirrors from the same mirror list. If the graph is cyclic graph, the topological sort will start
+from the first vertex after the alphabetical sorting of the vertces. If multiple verteces is from the same vertex, they will be
+visited in alphabetical order.
+
+3. The ImageStream has not supported pulling using tags yet. It has started working on this feature.
+
+### Risks and Mitigations
+
+Pulling images from mirror registries without the digest specifications could lead to returning different image version from different registry if the image tag mapping is out of sync. But the OpenShift core required image using digests to avoid different versions won't consume this feature at all, so it is not exposed to the risks that anyone who actually uses the feature will be exposed to.
+
+## Design Details
+
+### Test Plan
+
+Update the tests that are currently in the MCO to verify that registries.conf
+requires digest specification for mirrors configured through ImageSourceDigestPolicy.
+
+Update the tests that are currently in the MCO to verify that registries.conf does not
```

Comment:
Can you point me to these tests?

Reply:
The generated registry.conf  should be verified to not have restrictions about using tags with mirror
https://github.com/openshift/machine-config-operator/blob/3cc546103bc9042d899c34ada12e429b86aa6547/pkg/controller/container-runtime-config/container_runtime_config_controller_test.go#L583

Reply:
Thanks! Let's specific the container runtime config controller unit tests will be updated then.